vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -96,24 +96,40 @@ def _get_inputs_of_controls(action_id: ModelID, control_type: ControlType) -> Li
     ]
 
 
-def _get_inputs_of_chart_interactions(
+def _get_inputs_of_figure_interactions(
     action_id: ModelID, action_function: Callable[[Any], Dict[str, Any]]
-) -> List[State]:
-    """Gets list of States for selected chart interaction `action_name` of triggered page."""
-    chart_interactions_on_page = _get_matching_actions_by_function(
+) -> List[Dict[str, State]]:
+    """Gets list of States for selected figure interaction `action_name` of triggered page."""
+    figure_interactions_on_page = _get_matching_actions_by_function(
         page=_get_triggered_page(action_id=action_id),
         action_function=action_function,
     )
-    return [
-        State(
-            component_id=_get_triggered_model(action_id=ModelID(str(action.id))).id,
-            component_property="clickData",  # TODO: needs to be refactored to abstract implementation detail
-        )
-        for action in chart_interactions_on_page
-    ]
-
-
-def _get_action_callback_inputs(action_id: ModelID) -> Dict[str, List[State]]:
+    inputs = []
+    for action in figure_interactions_on_page:
+        triggered_model = _get_triggered_model(action_id=ModelID(str(action.id)))
+        if isinstance(triggered_model, Table):
```

Comment:
We probably want to move this logic to a Figure Model method called `_get_filter_interaction_inputs` or something similar. This is one way to avoid using statements like `if isinstance(triggered_model, Table)`.